### PR TITLE
 creating a data model

### PR DESCRIPTION
**Pull request by Grandi0z**
#### Changes made : 
- Generate migration for `User`, `Post`, `Comment`, and `Like`
- Set foreign keys in `posts`, `likes` and `comments`
- run the migration

### CODE REVIEW

File: db/migrate/20231003200705_rename_acolumn.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class RenameAcolumn < ActiveRecord::Migration[7.0]
+  def change
+    rename_column :posts, :user_id, :author_id
+  end
```
Comment (rloterh) [inline db/migrate/20231003200705_rename_acolumn.rb]: - [ ] Let's endeavor to properly establish migration rule for` Post` - `User` relations through `Author` like the example below:


```ruby
...
  def change
    add_reference :posts, :author, references: :users, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :posts do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Joseph, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 🌴  🏃 🌳 🏡 🏘️ 🌲 

## Highlights 👍  
- Good job so far establishing migration rule!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
Kindly endeavor to fix the broken logo link at the topmost part of your ReadMe
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20231003200705_rename_acolumn.rb (hunk 1 above)
Comment (Grandi0z) [reply db/migrate/20231003200705_rename_acolumn.rb]: Thank for the tips

Comment (rloterh) [reply db/migrate/20231003200705_rename_acolumn.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Joseph,  

**Blog app - creating data model** task is complete! 🏁 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on implementing recommended changes!
🎲 Professional ReadMe!

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**